yarn-project/p2p/src/mem_pools/tx_pool/aztec_kv_tx_pool.ts

```diff
@@ -193,14 +232,18 @@ export class AztecKVTxPool implements TxPool {
 
           if (!(await this.#minedTxHashToBlock.hasAsync(key))) {
             pendingCount++;
-            // REFACTOR: Use an lmdb conditional write to avoid race conditions with this write tx
-            await this.#pendingTxPriorityToHash.set(getPendingTxPriority(tx), key);
+            await this.addPendingTxIndices(tx, key);
             this.#metrics.recordSize(tx);
           }
         }),
       );
 
-      this.#metrics.recordAddedObjects(pendingCount, 'pending');
+      const { numLowPriorityTxsEvicted, numNewTxsEvicted } = await this.evictLowPriorityTxs(
+        hashesAndStats.map(({ txHash }) => txHash),
+      );
```

Comment:
@alexghr this function is reading from LMDB within a write tx. Will it be able to see what was written within this same tx? In other words, if we do:
```
store.write(1)
store.transaction(() => {
  store.write(2)
  value = store.read()
})
```

What will be the value of `value`?